midi_app_controller/gui/_tests/test_binds_editor.py

```diff
@@ -0,0 +1,300 @@
+import pytest
+from unittest.mock import patch
+from qtpy.QtWidgets import QPushButton
+from qtpy.QtCore import Qt
+from qtpy.QtTest import QTest
+
+from midi_app_controller.gui.binds_editor import ButtonBinds, KnobBinds, BindsEditor
+from midi_app_controller.models.controller import ControllerElement, Controller
+from midi_app_controller.models.binds import ButtonBind, KnobBind, Binds
+
+
+@pytest.fixture
+def controller_sample() -> Controller:
+    buttons = [
+        ControllerElement(id=1, name="Play"),
+        ControllerElement(id=2, name="Stop"),
+    ]
+    knobs = [
+        ControllerElement(id=3, name="Volume"),
+        ControllerElement(id=4, name="Zoom"),
+    ]
+
+    return Controller(
+        name="TestController",
+        button_value_off=0,
+        button_value_on=127,
+        knob_value_min=0,
+        knob_value_max=127,
+        buttons=buttons,
+        knobs=knobs,
+    )
+
+
+@pytest.fixture
+def button_binds_sample() -> list:
```

Comment:
Include type of the elements. You need to use `List` from `typing` to make it work in Python 3.8.